source/components/composing-components.md

```diff
@@ -0,0 +1,260 @@
+Components really shine when you use them to their full potential, that is, by composing them with one another.
+Take for example the `<ul>` element, and the fact that only `<li>` elements are appropriate as children.
+If we want the same type of behavior, then we have to compose our components.
+
+Just like we compose regular HTML elements, we can do the same with components.
+
+```app/templates/application.hbs
+{{#user-list users=activeUsers sortBy='name' as |user|}}
+  {{user-card user=user}}
+{{/user-list}}
+```
+
+### Component Blocks
+
+Components can be used in two forms, just like regular HTML elements.
+
+**Inline Form**
+
+```app/templates/application.hbs
+{{user-list users=activeUsers}}
+```
+
+This is the basic blockless form of a component, which doesn't give the
+consumer the ability to insert their own content into the given component.
+It's even possible that a component can only be used in this form.
+Make sure to consult a component's documentation for its available uses
+and forms.
+
+**Block Form**
+
+```app/templates/application.hbs
+{{#user-list users=activeUsers}}
+  {{!-- custom template here --}}
+{{/user-list}}
+```
+
+To compose components, we must use the block form, but we must also
+be able to distinguish from within our component which form is being used.
+This can be done with the `hasBlock` property from within our component.
+
+```app/templates/components/user-list.hbs
+{{#if hasBlock}}
+  {{yield}}
+{{else}}
+  <p>No Block Specified</p>
+{{/if}}
+```
+
+Once we know that the component is being used in block form, we can yield
+whatever the user has placed inside the block wherever we want via the `{{yield}}` helper.
+
+This `{{yield}}` helper can be used in the component's template definition multiple times,
+allowing us to create a component that works like a list, where the user insert/modify custom content
+at every yield. Take the following example, where we output a custom summary.
+
+```app/templates/components/post-list.hbs
+{{#each posts as |post|}}
+  <h3>{{post.title}}</h3>
+  <p>{{yield}}</p>
+{{/each}}
+```
+
+Which can be used like so:
+
+```app/templates/posts.hbs
+{{#post-list posts=headlinePosts}}
+  Greatest post ever!
+{{/post-list}}
+```
+
+And will result in the following HTML:
+
+```html
+<div id="ember123" class="ember-view">
+  <h3>Tomster goes to town</h3>
+  <p>Greatest post ever!</p>
+  <h3>Tomster on vacation</h3>
+  <p>Greatest post ever!</p>
+</div>
+```
+
+But what use is it to just output the same thing over and over? Don't we want to customize our posts,
+and display the right content? Sure we do. Lets explore the `{{yield}}` helper a bit.
+
+### Data Down
```

Comment:
From experience this depends if the data is isolated to your component, and if it is then go ahead and fetch the data from within your component. Otherwise, pass the data in. 

This also has to do with what kind of API you want the component to have.